framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4619,452 @@ private void makeGroundTargetType(
+      AnnotatedTypeMirror[] newTypeArgs = new AnnotatedTypeMirror[size];
+      Map<TypeVariable, AnnotatedTypeMirror> typeVarMap = new HashMap<>(size);
+      // Copy the non-wildcard type args from uncapturedType to newTypeArgs.
+      // If the non-wildcard type arg is type var add it to typeVarMap.
+      for (int i = 0; i < size; i++) {
+        AnnotatedTypeMirror uncapturedArg = uncapturedType.getTypeArguments().get(i);
+        if (uncapturedArg.getKind() != TypeKind.WILDCARD) {
+          AnnotatedTypeMirror copyArg = visit(uncapturedArg, originalToCopy);
+          if (copyArg.getKind() == TypeKind.TYPEVAR) {
+            typeVarMap.put(((AnnotatedTypeVariable) copyArg).getUnderlyingType(), copyArg);
+          }
+          newTypeArgs[i] = copyArg;
+        }
+      }
+
+      // Substitute the type variables in each type argument of capturedType using typeVarMap
+      // created above.
+      // This makes type variables in capturedType the same object when they are the same type
+      // variable.
+      for (int i = 0; i < size; i++) {
+        AnnotatedTypeMirror uncapturedArg = uncapturedType.getTypeArguments().get(i);
+        AnnotatedTypeMirror capturedArg = capturedType.getTypeArguments().get(i);
+        if (uncapturedArg.getKind() == TypeKind.WILDCARD) {
+          // Note: this if statement can't be replaced with if
+          // (TypesUtils.isCapturedTypeVariable(capturedArg))
+          // because if the bounds of the captured wildcard are equal, then instead of a captured
+          // wildcard, the type of the bound is used.
+          AnnotatedTypeMirror newCapArg =
+              typeVarSubstitutor.substituteWithoutCopyingTypeArguments(typeVarMap, capturedArg);
+          newTypeArgs[i] = newCapArg;
+        }
+      }
+      // Set captureType type args to newTypeArgs.
+      capturedType.setTypeArguments(Arrays.asList(newTypeArgs));
+
+      // Visit the enclosing type.
+      if (uncapturedType.getEnclosingType() != null) {
+        capturedType.setEnclosingType(
+            (AnnotatedDeclaredType) visit(uncapturedType.getEnclosingType(), originalToCopy));
+      }
+    }
+  }
+
+  /**
+   * Returns the list of type variables such that a type variable in the list only references type
+   * variables at a lower index than itself.
+   *
+   * @param collection a collection of type variables
+   * @return the type variables ordered so that eachtype variable only references earlier type
+   *     variables
+   */
+  public List<AnnotatedTypeVariable> order(Collection<AnnotatedTypeVariable> collection) {
+    List<AnnotatedTypeVariable> list = new ArrayList<>(collection);
+    List<AnnotatedTypeVariable> ordered = new ArrayList<>();
+    while (!list.isEmpty()) {
+      AnnotatedTypeVariable free = doesNotContainOthers(list);
+      list.remove(free);
+      ordered.add(free);
+    }
+    return ordered;
+  }
+
+  /**
+   * Returns the first TypeVariable in {@code collection} that does not contain any other type in
```

Comment:
I would change "contain" to "lexically contain" (if that is true!) to distinguish from the JLS notion of type containment.